sdk/storage/azure-storage-blob/azure/storage/blob/_shared/shared_access_signature.py

```diff
@@ -0,0 +1,368 @@
+             get_value_to_append(_QueryStringConstants.SIGNED_IP) +
+             get_value_to_append(_QueryStringConstants.SIGNED_PROTOCOL) +
+             get_value_to_append(_QueryStringConstants.SIGNED_VERSION))
+
+        if service in ['blob', 'file']:
+            string_to_sign += \
+                (get_value_to_append(_QueryStringConstants.SIGNED_CACHE_CONTROL) +
+                 get_value_to_append(_QueryStringConstants.SIGNED_CONTENT_DISPOSITION) +
+                 get_value_to_append(_QueryStringConstants.SIGNED_CONTENT_ENCODING) +
+                 get_value_to_append(_QueryStringConstants.SIGNED_CONTENT_LANGUAGE) +
+                 get_value_to_append(_QueryStringConstants.SIGNED_CONTENT_TYPE))
+
+        # remove the trailing newline
+        if string_to_sign[-1] == '\n':
+            string_to_sign = string_to_sign[:-1]
+
+        self._add_query(_QueryStringConstants.SIGNED_SIGNATURE,
+                        _sign_string(account_key, string_to_sign))
+
+    def add_account_signature(self, account_name, account_key):
+        def get_value_to_append(query):
+            return_value = self.query_dict.get(query) or ''
+            return return_value + '\n'
+
+        string_to_sign = \
+            (account_name + '\n' +
+             get_value_to_append(_QueryStringConstants.SIGNED_PERMISSION) +
+             get_value_to_append(_QueryStringConstants.SIGNED_SERVICES) +
+             get_value_to_append(_QueryStringConstants.SIGNED_RESOURCE_TYPES) +
+             get_value_to_append(_QueryStringConstants.SIGNED_START) +
+             get_value_to_append(_QueryStringConstants.SIGNED_EXPIRY) +
+             get_value_to_append(_QueryStringConstants.SIGNED_IP) +
+             get_value_to_append(_QueryStringConstants.SIGNED_PROTOCOL) +
+             get_value_to_append(_QueryStringConstants.SIGNED_VERSION))
+
+        self._add_query(_QueryStringConstants.SIGNED_SIGNATURE,
+                        _sign_string(account_key, string_to_sign))
+
+    def get_token(self):
+        return '&'.join(['{0}={1}'.format(n, url_quote(v)) for n, v in self.query_dict.items() if v is not None])
+
+
+class BlobSharedAccessSignature(SharedAccessSignature):
+    '''
+    Provides a factory for creating blob and container access
+    signature tokens with a common account name and account key.  Users can either
+    use the factory or can construct the appropriate service and use the
+    generate_*_shared_access_signature method directly.
+    '''
+
+    def __init__(self, account_name, account_key):
+        '''
+        :param str account_name:
+            The storage account name used to generate the shared access signatures.
+        :param str account_key:
+            The access key to generate the shares access signatures.
+        '''
+        super(BlobSharedAccessSignature, self).__init__(account_name, account_key, x_ms_version=X_MS_VERSION)
+
+    def generate_blob(self, container_name, blob_name, permission=None,
```

Comment:
I feel the name is confusing, it seems the method is to generate a token not a blob?

Reply:
Thanks! These are private/internal methods, copied from the track1 SDK. I will look at refactoring them/rename as needed for a subsequent preview!